blockchain/sync_manager/src/sync.rs

```diff
@@ -78,32 +74,25 @@ impl<'a> Syncer<'a> {
         Ok(())
     }
     /// Returns message root CID from bls and secp message contained in the param Block
-    fn compute_msg_data(&self, block: &Block) -> Result<Cid, CidError> {
+    fn compute_msg_data(&self, block: &Block) -> Result<Cid, Error> {
+        // retrieve bls and secp messages
         let bls_msgs = block.bls_msgs();
         let secp_msgs = block.secp_msgs();
-
-        let mut bls_cids = Vec::new();
-        for bls_msg in bls_msgs {
-            bls_cids.push(bls_msg.cid());
-        }
-
-        let mut secp_cids = Vec::new();
-        for secp_msg in secp_msgs {
-            secp_cids.push(secp_msg.cid());
-        }
-
-        let mut amt = AMT::new(&self.chain_store.db);
+        // collect bls and secp cids
+        let bls_cids = bls_msgs.iter().map(|x| x.cid()).collect::<Vec<_>>();
+        let secp_cids = secp_msgs.iter().map(|x| x.cid()).collect::<Vec<_>>();
+        // generate AMT and batch set message values
         let bls_root = AMT::new_from_slice(&self.chain_store.db, &bls_cids)?;
         let secp_root = AMT::new_from_slice(&self.chain_store.db, &secp_cids)?;
 
         let meta = MsgMeta {
             bls_message_root: bls_root,
             secp_message_root: secp_root,
         };
+        // store message roots and receive meta_root
+        let meta_root = self.chain_store.db.put(&meta)?;
```

Comment:
Just a comment to consider changing the db type to an implementation of the BlockStore trait, since that is the only way it is used, but I can explore this in a seperate PR